### PR TITLE
[BEAM-1047] Add getRegion to DataflowPipelineOptions

### PR DESCRIPTION
**Pull request by dhalperi**

### CODE REVIEW

**Comment by dhalperi** [comment]
R: @tgroh or @kennknowles

File: runners/google-cloud-dataflow-java/src/test/java/org/apache/beam/runners/dataflow/DataflowRunnerTest.java [hunk 1]
```diff
@@ -304,16 +309,21 @@ public void testRun() throws IOException {
...
+   * Options for testing.
+   */
   public interface RuntimeTestOptions extends PipelineOptions {
+
     ValueProvider<String> getInput();
+
     void setInput(ValueProvider<String> value);
 
     ValueProvider<String> getOutput();
+
```
Comment (dhalperi) [inline runners/google-cloud-dataflow-java/src/test/java/org/apache/beam/runners/dataflow/DataflowRunnerTest.java]: I will manually revert some autoformat carnage. sorry.

File: runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/options/DataflowPipelineOptions.java [hunk 2]
```diff
@@ -97,6 +97,23 @@
...
   /**
+   * The Google Compute Engine
+   * <a href="https://cloud.google.com/compute/docs/regions-zones/regions-zones">region</a>
+   * for creating Dataflow jobs.
+   *
+   * <p>NOTE: The Cloud Dataflow service does not yet honor this setting. However, once service
+   * support is added then users of this SDK will be able to control the region.
+   */
+  @Hidden
+  @Experimental
```
Comment (tgroh) [inline runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/options/DataflowPipelineOptions.java]: Experimental type?

File: runners/google-cloud-dataflow-java/src/test/java/org/apache/beam/runners/dataflow/DataflowPipelineJobTest.java [hunk 3]
```diff
@@ -359,21 +375,21 @@ public void testGetAggregatorValuesWithNoMetricUpdatesReturnsEmptyValue()
...
-        appliedPTransform(fullName, pTransform, TestPipeline.create());
+        appliedPTransform(fullName, pTransform, Pipeline.create(options));
 
     DataflowAggregatorTransforms aggregatorTransforms = new DataflowAggregatorTransforms(
         ImmutableSetMultimap.<Aggregator<?, ?>, PTransform<?, ?>>of(aggregator, pTransform).asMap(),
         ImmutableMap.<AppliedPTransform<?, ?, ?>, String>of(appliedTransform, stepName));
 
     GetMetrics getMetrics = mock(GetMetrics.class);
-    when(mockJobs.getMetrics(PROJECT_ID, JOB_ID)).thenReturn(getMetrics);
+    when(mockJobs.getMetrics(PROJECT_ID, REGION_ID, JOB_ID)).thenReturn(getMetrics);
```
Comment (tgroh) [inline runners/google-cloud-dataflow-java/src/test/java/org/apache/beam/runners/dataflow/DataflowPipelineJobTest.java]: More out of curiosity than anything else - do you know if there's a compelling reason to use the mockito matchers here? our styling is inconsistent

Comment (dhalperi) [inline runners/google-cloud-dataflow-java/src/test/java/org/apache/beam/runners/dataflow/DataflowPipelineJobTest.java]: if all are constants, prefer no `eq()`.

**Pull request closed without merging**